overlord/devicestate/devicemgr.go

```diff
@@ -159,6 +161,8 @@ func maybeReadModeenv() (*boot.Modeenv, error) {
 
 // StartUp implements StateStarterUp.Startup.
 func (m *DeviceManager) StartUp() error {
+	secboot.HasFDERevealKey = hasFDERevealKey
```

Comment:
This cannot be done here, secboot.HasFDERevealKey is used by snap-bootstrap and that cannot import all the manager. So we need to create a devicestate/fde package that does not import anything from managers and that is imported by secboot. Also move the JSON structs there.